### ISSUE TITLE
License Change: Please explain

### ISSUE THREAD

**Issue by cookiengineer** [aseprite/aseprite#1242]
You changed the license in this commit:
https://github.com/aseprite/aseprite/commit/5ecc356a41c8e29977f8608d8826489d24f5fa6c

It would be awesome to hear why you did that and what the reason was. There's a negative discussion going on on reddit regarding GPL work from other contributors of this project. I think it's important to explain things, because there might be a shitstorm rising about this project - which I personally would think would be a shame.

I'm just here to let you know about it, would be bad to see a shitstorm about this awesome project, so please go explain things on reddit to let people understand the reasons.

Link to reddit Discussion:
https://www.reddit.com/r/opensource/comments/50n2pl/pixel_art_editor_aseprite_just_switched_from_gpl/

Warm Regards,

**Comment by dacap** [comment]
Hi @cookiengineer, thanks for creating this. I'm writing a blog post about this and I'll post the link here soon.

**Comment by dacap** [comment]
I've just posted comments about this in the Aseprite devblog: http://dev.aseprite.org/post/149797781837/new-source-code-license

**Comment by cookiengineer** [comment]
Thanks much!

**Comment by davexunit** [comment]
Aseprite ought to be forked so the ethical, freedom-respecting version can continue to be maintained by a community, rather than see its users oppressed by its developer.

**Comment by dacap** [comment]
@davexunit actually there is already a fork in https://github.com/aseprite-gpl/aseprite and sorry to make you feel oppressed 😢

**Comment by davexunit** [comment]
Great, I will change the package I maintain to use this fork.  So long!

**Comment by dacap** [comment]
@davexunit yes, I think Linux distributions could start using the new fork (I think I might help in the fork in some near future too, e.g. to keep compatibility between `.ase` file format). Please remember my advices in the blog post: http://dev.aseprite.org/post/149797781837/new-source-code-license

**Comment by dacap** [comment]
As a side note, I've just sent an email to contributors in case they want their changes outside this repository. I know I could use the CLA to sublicense the changes, but the CLA wasn't originally made for this (my original idea was just to create an iPhone/iPad version, because the Apple Store doesn't allow GPL code).

**Comment by MarcoEstevez** [comment]
And here it is, the work of your entire live got cloned and is not yours anymore...

As a windows user that pays for any software (binaries not sources) I use, I have never got the whole point with free software thing, but reading this thread and the one on reddit and also @dacap blog, I just saw how violent "openSource world" can be. and makes me appreciate a bit more paid fair software.

Poor oppressed users, may I complaint to my electrical company, cause they have been oppressing me for lot of years till now?

BTW: @dacap congrats for your product, I love it.

**Comment by dacap** [comment]
Thanks for your words @MarcoEstevez

**Comment by zed9h** [comment]
I am a recent contributor, and will continue to contribute, even without GPL, because the CLA is not that bad, and @dacap is doing such a good work as a maintainer and designer, that I don't see any reasonable alternative.

But @MarcoEstevez, let's not forget the purpose of the open source licenses: imagine if ffmpeg or linux got pulled off by the maintainer, after thousands of contributors giving away their work for years to the public benefit. It's specially true it multiple enterprises are contributing to a common project..

If we use the argument about the electric bill literally, the maintainer would have to pay for all contributions, and all the discussion about common good could be thrown out of the window.

**Comment by oceanhahn** [comment]
You know, I can appreciate how these shitstorms get going? Developers start these projects as labours of love, and throw some boilerplate license on it. Keeners add their bit, it gets rolled together, and turns into something more than it might have been otherwise. But that's the sort of development base it has: keeners. It's a rare thing that a developer also enjoys disassembling the fine, exploitive nuance of copyright law at all, let alone at the outset of a personal-project-cum-open-source-product. And because the thing called "intellectual property" is such a racid topic among enthusiasts, even the slightest whisper of it can trigger an overblown, paranoid reaction.

Users feel as though something of theirs is being stolen from them-- if they've also contributed to the project, they may feel this way with good reason. The nature of Open Source projects can create a sense of security that only sometimes really exists. Some others may resent the very thought of an open source franchise moving away from the movement, even slightly, without having even the slightest investment in that product to begin with. The vast majority of users never notice-- when was the last time anyone actually _read_ their software licensing agreement? To the end? (If you raised your hand, I admit that I am impressed.) They have a right to be paranoid since property theft is a crime perpetrated by gigantic multinationals and super-smalltimers alike. But their reactions can be quite extreme in less-graceful instances (n.b.: Nobody in this case, that I've seen, has been so awful, just for the record).

Developers, especially hobbyist developers, on the other hand, usually (*not always!*) just want to try to keep the scope of a growing project under control and to maybe try and capitalize on their efforts a little better, and in these cases, it comes down the attitude of the project administrators. In this case, I trust @dacap to not go full-Ebaum's on this one (ha ha how is that for a dated reference? :v ), but I can completely appreciate how others would not be so generous. 
I've also paid my few dollars, so I don't have as much skin in the game. :I

In any case, if you ask me (and nobody did), the only real misstep here was not mentioning that the license would be changing some time in advance. License management for any open-source project is not some tepid swimming pool to dip your toe into, it is a howling acid waterfall that drops you off a cliff and sucks you under. The more you prepare for it, the less chaotic and unpredictable it becomes. Somewhat.
It's better to sort out sooner and with as much collaboration as possible, and there are things we cannot imagine anyone caring about turning into huge stumbling blocks. "The best time to plant a tree is twenty five years ago. The second best time is now."

Anyhoo, it's nice to see a project grow from humble beginnings, and it's not like the Open Source splinters will vanish. I hope both the EULA and GPL versions both thrive and grow; it's not like there isn't room enough for them both.

We're all on the same side here.

**Comment by oceanhahn** [comment]
Woah ha ha o wow that was way more lecturing than I had intended it to be. x_x

*tl;dr:* It seems to me more reasonable to think of the open-source version losing @dacap as a contributor than to think of the entire ASEprite project vanishing from the realm of open-source'dness.
The license change probably should have been given with a bit of notice, though. Hindsight.
Looking forward to more updates.

**Comment by oceanhahn** [comment]
@zed9h I feel like the electric company analogy was a response to the rather melodramatic rights-trampled-upon remark by @davexunit. Probably you know that, though?

**Comment by zed9h** [comment]
@oceanhahn, there is no oppression here, the GPL grants enough protection to equalize the sides and make the rules of the game pretty clear. Now I am taking some risk here, being not protected by it, but the CLA is reasonable, as I said, and I am trusting @dacap, because he is doing this great work so consistently for such a long time. I also don't feel it's a crime to try novel ways to earn a living with his work, that, as you put it, is clearly a work of love and I also respect the fact the he is still keeping the source open.

**Comment by oceanhahn** [comment]
@zed9h We're agreed, then! :smiley:

**Comment by oceanhahn** [comment]
@cookiengineer Makes sense to me, but it's not my project we're talking about here. ;) 
What does @dacap think?

**Comment by eska014** [comment]
The discussion about this situation is all over the place, so I guess this ticket is as good a place as any to add some comments.

- The flaming written by some writers from both sides is pathetic.
This includes angry open source advocates. It also includes feeding those emotions with attacks against the open source community per guilt by association. Insults, zealous accusations and sarcasm are only making this issue worse than it already is

- The license change process lacked in transparency.
It shouldn't have been an issue to make the change along with the blog article, a Twitter message, or anything.
Instead I heard about the change in an unrelated IRC channel by word of mouth, days after. No one even seems to have been aware of the change at that point. A blog announcement followed only after the drama started.
There's still misunderstandings about the concrete issue that lead to this change, see several people taking about having to 'pay the bills' or 'earning a living' above.

- Aseprite was the best open source pixel art software.
Even in comparison with pixel art tools in general, Aseprite continues to be one of the best. But the loss of freedom is a loss of one of the big pluses of the software

- Going proprietary seems like an excessive reaction to a potentially fixable problem.
It's great that @dacap communicates and details a problem with software distribution and feedback, but changing the licensing to 'fix' it doesn't make sense to me. What communication was there to solve this problem before creating this frustrating licensing situation?

**Comment by MattGuerrette** [comment]
From the statements of @dacap  this software should never have been GPL to begin with.

I just feel bad for those who contributed. At least you all can continue on the fork.

**Comment by tony** [comment]
@MattGuerrette bingo

See all this drama caused?  See how @dacap, the guy who did >95% of the work getting mobbed?

Look at this: https://github.com/aseprite/aseprite/graphs/contributors, just 27 contributors. Over all those years.

The moral of the story, what this message sends: pick GPL, build your project in good faith, and be [prepared to get floored](https://www.reddit.com/r/opensource/comments/50n2pl/pixel_art_editor_aseprite_just_switched_from_gpl/).

What a backwards way to treat people who devoted a good deal of their life to a project.  Can you imagine for a second how it feels?

In many cases, the people complaining aren't contributors at all.

People who keep talking about "freedom". As a fellow open source programmer who used to cargo cult GPL longer than a decade ago, I want you to ask confront honestly what GPL gives you and what you lose.

1. See how permissive licenses end up getting you the same net-result in the end (seeing the source, contributions back), examples: I can build a memstick image in freebsd right now
2. See how GPL's side effects complicate projects - LLVM and Clang, Linux and ZFS. I've also probably helped 10 or so projects convert from (L)GPL to permissive licenses, the gray hairs involved.. by the time the conversion was complete I didn't require the library anymore 3/4 of the time.

Let this be a lesson to people who pick GPL, the values it espouses are great, but the license in itself has so many side effects that burn projects in the long run.

I'm fine with any license change @dacap wants.

**Comment by objectbakery** [comment]
Hey @decap 

a very comprehensible justification at your dev blog!

Its embarrassing an humiliating how people behave here (@davexunit) that seems symptomatic for the open source scene. 'Oppressing users'?? WTF !?!?!

I will stay with Aseprite! They can fork the software but they can not fork your ideas and your commitment. All the best! Looking forward for future releases!

**Comment by winterheart** [comment]
Hello.
I'm one of contributors to Aseprite. I just want to say that I'm fine with license change (though it means that I cannot maintain package in my favorite Gentoo Linux distributive). As contributor I signed CLA so there no questions about legal issues.
Still I disagree that Linux users do not contribute at all. As Linux user, I made some small improvements in build subsystem to make benefits to all platforms.

**Comment by dacap** [comment]
> the only real misstep here was not mentioning that the license would be changing some time in advance

Hi @oceanhahn, you are right. It was my mistake to not ask contributors about the license change.

> Why not have a dual licensed model in aseprite with a popup on first start?
> - GPL for free software users that want to create an open source game.
> - EULA and paid license option for guys that want to build a paid game or app with it.

Hi @cookiengineer, that seems like a good idea. Anyway as I said in my post, I don't agree anymore with the GPL.

> Going proprietary seems like an excessive reaction to a potentially fixable problem.

Hi @eska014, I agree with everything you said. The only thing is that I've changed my mind about GPL in these days. And the problems are described in the post in the "So why not GPL?".

> Still I disagree that Linux users do not contribute at all. As Linux user, I made some small improvements in build subsystem to make benefits to all platforms.

Hi @winterheart, I think that Linux programmers (let say, Linux users that can program) are of great help. They are the best. The issue I'm seeing is that Linux distributions do not make a way that regular end-users (let's say non-programmers) can give useful information to fix problems.

And finally thanks to all the kind words.

**Comment by jarfil** [comment]
Hi @dacap, I found your blog post very interesting, but you completely lost me at the "So why not GPL?" part.

I mean, you first talk about "If someone doesn’t want to (or just cannot) pay for your work, she/he is not a customer", and then suddenly you turn around and say "GPL [...] Users have a responsibility, they should be encouraged to contribute". But you earlier realized that "your software is not a service", so why care about those users at all, who are not programmers and will not contribute to your work anyway?

By this I'm not trying to say what you should or shouldn't do, it's your code and you do whatever you want, but the reasoning seems contradictory.

**Comment by Download** [comment]
> And here it is, the work of your entire live got cloned and is not yours anymore...

Don't worry. Nothing will happen. Cloning a project is one press of the `Fork` button. Actually adding something of value to it is a whole different story. Just wait and see how that fork is going to not do anything but sit there.

> pick GPL, build your project in good faith, and be prepared to get floored.

This is so true. The GPL as a license sucks imho.

Why?
Because the GPL has gone too far the other way. And because it's a license that's based on fear instead of love. And because it explicitly restricts derivatives into using the GPL, making it incompatible with most other OSS licenses. 

Especially the fear thing. The whole license oozes a fear that someone else might end up making money on GPL'd software. There are needless restrictions against the authors of (derivatives of ) the work. MIT, BSD, ASF etc licenses do not have these restrictions, yet they thrive.

Take the app store. GPL advocates act as if Apple has something against the GPL or OSS in general, when in fact it is restrictions in the GPL itself that make it essentially incompatible with the app store. And FSF people were threatening to sue, almost forcing Apple into it's current position. Why??? Having followed the FSF and GPL for years, though I respect them for their great early role in OSS, I now feel that they are actually hurting it. No GPL software on the app store isn't good for users. But it's a result of the fears included in the license text itself.

@dacap Keep up the great work and do with *your* baby what you want!

Oh and one more thing. The GPL never promises that there will be new versions of the software. Not choosing to publish future versions of the software under the GPL isn't taking anything away from anyone!
How so?
It's gone too far in protecting users against a threat that isn't real, whereas at the same time it has sacrificed the rights and protections of those actually contributing. Why is it t

**Comment by dacap** [comment]
> so why care about those users at all, who are not programmers and will not contribute to your work anyway?

Hi @jarfil, in that case users at least know that "there is a cost." It's not just "free" to get the program. You pay for it, or you compile it, or you get it in some "ilegal" way (which is another kind of "cost" in case he/she wants to take the risk). Linux distributions remove that "cost" completely, giving the sensation that software is like something "disposable" (no cost, no consequences). Thanks for sharing your thoughts.

> @dacap Keep up the great work and do with your baby what you want!

Hi @Download, thanks for your kind words!

**Comment by est31** [comment]
@Download The current EULA of this project and most proprietary licenses are far more restrictive than the GPL is. If you don't like the GPL because of its restrictions, you shouldn't like proprietary licenses either.

About this whole license change: @dacap is the author of this program, and its his decision whether he wants to keep further work under a proprietary license or not. It still makes me sad that his present and future work won't be made available as free software.

He seems to blame the distributions for this, and its sad if distros couldn't create a more friendly environment for people like him, so that more chose to stay on a free software path. Donation buttons, or stuff like that, these are very small things! The F-Droid app store has such functionalities.

**Comment by Download** [comment]
His present work is already available under the GPL and that won't (can't) change. His new work will also be available *for free*. The GPL can't claim sole ownership of 'freedom' (though I know that's what they like to do). Yes, software can exists that is *free software*, even though it's not licensed under the GPL.  Aseprite is now yet another example of such software.

Want to keep it GPL? Easy. Contribute to the GPL fork.

But that's the not happening part. The number 2 contributor to this project made just 25 commits, versus dacap's 4485 commits. The guy that forked this project, @akien-mga, made exactly ZERO commits to this project in it's entire history. He made ONE commit to the fork... the one that restores the GPL license... I wouldn't count on him to keep the project alive. All this 'keep it free' talk is nice, but doing some actual work to help the project is a lot nicer.

**Comment by akien-mga** [comment]
@Download The strong unbalance in the contributors graph is IMO likely due to the CLA. Even though I understand the (IMO pertinent) reasons for having a CLA in the first place, the added bureaucracy and risk of relicensing (like it happened) would likely put off many potential contributors, me included. Being open source does not automagically generates a community of co-developers that will do tons of pull requests, the project itself must be capable of integrating such changes and to foster the various personal motivations that lead to contributing - many developers contribute for the love of free software, and would therefore avoid a project with a CLA that could lead to their work being relicensed under a proprietary EULA. There are of course other motivations, and many CLA-signers already stated that the license change does not bother them and that they will continue to contribute, which is awesome.

So no, I was not an aseprite contributor as far as code goes. My contribution was to [package aseprite for the Mageia distribution](http://svnweb.mageia.org/packages/cauldron/aseprite/current/SPECS/aseprite.spec?view=markup), and to try to do this well.

Please read https://github.com/aseprite-gpl/aseprite/issues/1 as to my motivation for creating a fork, I never mentioned that I intended myself to lead the development of a competitive fork.

Yes, I can keep the GPL fork alive as in "fix some simple bugs and ensure that the code still builds and runs fine for the years to come", but no, I don't claim that I can produce a *better* aseprite than @dacap. @dacap created and wrote most of aseprite, and this software is so successful because he has a great vision and strong attention to the workflow - as time goes the proprietary aseprite should become even more awesome, and unless some very dedicated contributors arise, the GPL fork should stall around the current state (which is already a great piece of software IMO, and thus worth preserving).

**Comment by Download** [comment]
> Why not have a dual licensed model in aseprite with a popup on first start?

> * GPL for free software users that want to create an open source game.
> * EULA and paid license option for guys that want to build a paid game or app with it.

Unfortunately, this is not possible. This is one of the reasons why the FSF sucks. They keep saying 'free as in free speech, not as in free beer', but actually, if you study the license, it *is* free as in free beer. There is no way around it. They make you think you can charge money for the software when actually you can't. 

The GPL, just like all other Open Source licenses, does not allow *discrimination* against groups of users. If you make software available under the GPL license then it's available for anyone that accepts the terms in it. Even if they are EA and are making millions. At least the other licenses don't make you think you can somehow keep control.

The only way to make money off of GPL software is by dual licensing it. This can only be voluntary (because of the no discrimination rule), so you need customers that have a reason to want to do this. Because the only major restriction in the GPL is the one of reciprocity (quid for quo, if you make derivatives, these too need to be licensed under the GPL) it mostly means you will be targeting 'customers' trying to make a different version of Aseprite itself which they don't want to make available under the GPL... probably not a very big market.

**Comment by Download** [comment]
> The strong unbalance in the contributors graph is IMO likely due to the CLA. 

Nah. This is actually a very typical distribution for most open source projects. They are mostly either one-man-shows, or backed by some corporation that just has staff maintaining it. Linux is a big exception. There are thousands of projects with this distribution graph.... Actually, this one is pretty good with 27 contributors. Most projects have far less.

**Comment by Download** [comment]
Apart from the fact that your fork is most likely in fact, illegal, it's also not needed at all.

> Since https://github.com/aseprite/aseprite went proprietary on Aug 26th, 2016 (aseprite@5ecc356), this fork and corresponding GitHub organization were created to preserve the last GPLv2 version of the code.

There is no need to 'preserve' the last GPLv2 version, because that's exactly what Git was built for. Need the last GPLv2 version? [Here it is](https://github.com/aseprite/aseprite/releases/tag/v1.1.7). No fork needed.

I say your fork is probably illegal because what you did was fork the master, which is now and was when you made the fork, licensed under @dacap 's eula and not under the GPL. Then you broke that eula when you made your changes.

What you should have done was base your fork on v1.1.7. You would also not have to make any changes to the license then. because 1.1.7 is GPL licensed to begin with. 

But actually, you didn't need to make the fork to begin with. because the last GPL version is safe, right here on GitHub. And there were already 188 forks of this project, so I'd say it was pretty safe already.

**Comment by akien-mga** [comment]
@Download Check your facts, I did not fork the master, but the last GPLv2 commit before the license change. Then I removed the EULA file which does not apply to the source code as distributed under the GPLv2 license. And no, my point is not just to "preserve" the code as if it were a static tarball (tarballs exist already for that...), but to give a place for the contributors interested in a GPLv2 version to contribute to. Again, read the above mentioned issue, and check the commit history...

**Comment by Download** [comment]
@akien-mga I stand corrected.

**Comment by dacap** [comment]
People, I'd like to say a few things:

1. To thank you all for the conversation.
1. I think that @akien-mga has the best intentions with the fork, so I hope both projects can continue with their objectives.
1. I just want to program software, I'm not sure when all this become so bureaucratic but it sucks, I'm not a lawyer and I really think I would lose any kind of trial no matter what I do.
1. I've stopped to watch `aseprite-gpl` because I don't want to be influenced by their decisions or patches, please @akien-mga (or someone in the project) send me a line in case you think that some issue needs my attention.

**Comment by Silent-Hunter** [comment]
My only comment is to say that this license change will kill the ability to use this project on Linux handhelds and other such devices that have no built in way to compile code. They need to be able to distribute it as a binary, or none of their users can use it. Unless you want to provide a paid app for ARM Linux.

**Comment by dacap** [comment]
@Silent-Hunter hi Silent Hunter, I'm thinking that I might offer a "distribution" exception for platforms that are completely unsupported. Actually I'm doing that for schools or institutions that want to redistribute Aseprite freely in their classroom.

**Comment by Silent-Hunter** [comment]
Oh! That's a good idea!

**Comment by jarfil** [comment]
@Download The GPL is essentially a "software support business protection license", no more, no less.

You write a piece of software, give it to someone, and get paid for support. If at any point you stop supporting it, you stop getting paid. Which is when anyone else can step in and offer a better support than you do... but, at the same time, can NOT use your code to build something that would lock you our from supporting that same client if they later on decided to switch back to you. You also get all the code written by your competing support businesses, whether your client wants to share it with you, or if they publicly share it as a competitive advantage to distinguish themselves.

If your business model is not software support, then there is little reason to use the GPL. On the other hand, if it is software support, then there is little reason not to use it.

**Comment by Download** [comment]
I'd say that even if your business is in providing support (for code you wrote), the GPL still is very tricky at best and damaging at worst. The biggest problem is that it's so difficult to do anything else with the code. Once you slap a GPL license on it and start accepting contributions, you basically give up all your rights as a copyright holder to do something else with that code should you later desire. Had you chosen, say, the MIT license, you could do whatever you wanted to with it. And so could others.

What if some client decides he wants to include your code in a bigger product? You will have to inform them that he has to make his product available under the GPL... which he probably won't like. You end up in a situation where you no longer control what you do with your own work... even if you made over 4000 commits and the second largest contributor only made 25 commits. Too bad for you. 

Why risk that? To prevent others from making millions off of your code? I don't see the advantage.
